cylc/flow/remote.py

```diff
@@ -202,24 +202,27 @@ def construct_rsync_over_ssh_cmd(
     rsync_cmd.append("--exclude=*")  # exclude everything else
     rsync_cmd.append(f"{src_path}/")
     rsync_cmd.append(f"{dst_host}:{dst_path}/")
-    return rsync_cmd
+    return rsync_cmd, dst_host
 
 
-def construct_ssh_cmd(raw_cmd, platform, **kwargs):
+def construct_ssh_cmd(
+    raw_cmd, platform, badhosts=None, **kwargs
+):
     """Build an SSH command for execution on a remote platform.
 
     Constructs the SSH command according to the platform configuration.
 
     See _construct_ssh_cmd for argument documentation.
     """
+    host = get_host_from_platform(platform, badhosts=badhosts)
     return _construct_ssh_cmd(
         raw_cmd,
-        host=get_host_from_platform(platform),
+        host=host,
         ssh_cmd=platform['ssh command'],
         remote_cylc_path=platform['cylc path'],
         ssh_login_shell=platform['use login shell'],
         **kwargs
-    )
+    ), host
```

Comment:
I wonder if it would be better to make `host` an arg of `construct_ssh_cmd()`, replacing `badhosts`. Then `construct_ssh_cmd()` wouldn't have to return the host as well as the cmd

Reply:
It might well be. I'll have a look. (I think that this is putting things back to how they were before platforms...)

Reply:
I have done this